### PR TITLE
[VIP-Travel] Trip room summary

### PR DESCRIPTION
**Pull request by rushatgabhane**
### Fixed Issues

$ https://github.com/Expensify/App/issues/37825
PROPOSAL:

### Tests

1. Go to oldDot - staging.expensify.com
2. Go to travel
3. Book a flight, rail, car and hotel.
4. Go to newDot and go to the trip room.
5. Verify that you see trip room summary, and that it is as per [figma](https://www.figma.com/file/rOyUSHbLLyp64emz6WeoPW/Travel?type=design&node-id=157-18561&mode=design&t=G1T5wHEFJMo34pq0-0)
image

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

image

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

image

image

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Backend having some performance issues, so im not getting pusher updates. 
</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

image

image

image

image

image

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by shawnborton** [comment]
This spacing from the first part of your PR comment looks good:
image


But further down in your screenshots section, the spacing here looks different:
image

**Comment by rushatgabhane** [comment]
@shawnborton thank you! Let me fix it

**Comment by rushatgabhane** [comment]
@shubham1206agra waiting on newDot to receive backend data - https://expensify.enterprise.slack.com/archives/C05S5EV2JTX/p1715299637326109?thread_ts=1714773700.569819&cid=C05S5EV2JTX

**Comment by rushatgabhane** [comment]
image

**Comment by shawnborton** [comment]
It's still not quite right. We are using 4px gaps between the three lines. The top line should be 16px tall since our label font uses a 16px line height. The middle line should be 20px tall since our regular font uses a 20px line height. And the bottom line should be 14px tall since our micro font size uses a 14px line height.

**Comment by rushatgabhane** [comment]
1. Top line `lineHeight: 16`
image


2. Middle `lineHeight: 20`
image


3. Bottom line: `lineHeight: 14px`
image


Fixed the gap to be 4px.

image

**Comment by shawnborton** [comment]
Much better, thank you!

**Comment by rushatgabhane** [comment]
Updates here - https://expensify.slack.com/archives/C05S5EV2JTX/p1715723531626149

**Comment by rushatgabhane** [comment]
Waiting on car details

**Comment by rushatgabhane** [comment]
@shubham1206agra this is ready for your review. There are a few backend issues [here](https://expensify.slack.com/archives/C05S5EV2JTX/p1716647047009049) and [here](https://expensify.slack.com/archives/C05S5EV2JTX/p1716641603874949?thread_ts=1716570830.369449&cid=C05S5EV2JTX) but it shouldn't block anything.

**Comment by rushatgabhane** [comment]
With the data available from backend, this is what we have now - 

image

**Comment by rushatgabhane** [comment]
@shubham1206agra any idea what is causing the lint issues? I can't figure it out.

image

File: src/types/onyx/OriginalMessage.ts
Comment (rushatgabhane) [reply src/types/onyx/OriginalMessage.ts:350]: actually, im not sure if we even need this.

Comment (rushatgabhane) [reply src/types/onyx/OriginalMessage.ts:350]: no that wasn't it. the const was not updated.

Comment (rushatgabhane) [reply src/types/onyx/OriginalMessage.ts:350]: I don't know how to find the type for the original message. What does original message even mean?

Comment (rushatgabhane) [reply src/types/onyx/OriginalMessage.ts]: updated as per - https://expensify.slack.com/archives/C05S5EV2JTX/p1716853395803129?thread_ts=1716853202.001619&cid=C05S5EV2JTX

**Comment by rushatgabhane** [comment]
@shubham1206agra this PR is ready again. All tests pass, please review. 

The backend [enhancements](https://github.com/Expensify/App/pull/41659#issuecomment-2131309897) would be fixed soon.

**Review by shubham1206agra — APPROVED** [review]

**Review by shubham1206agra — APPROVED** [review]

**Comment by amyevans** [comment]
@stitesExpensify want to review this one?

**Comment by stitesExpensify** [comment]
Sure I can take a look. An extra review from you would also be appreciated though 🙇

File: src/libs/DateUtils.ts
Comment (stitesExpensify) [inline src/libs/DateUtils.ts:772]: This is backwards right? We want to not display the year if it is this year (below) and not show the year in this case?

**Review by stitesExpensify — CHANGES_REQUESTED** [review]
LGTM! Only found one change

File: src/components/ReportActionItem/TripDetailsView.tsx
Comment (amyevans) [inline src/components/ReportActionItem/TripDetailsView.tsx:126]: We should use `{shouldUseNarrowLayout} = useResponsiveLayout()` now per https://expensify.slack.com/archives/C02NK2DQWUX/p1717010005094209

File: src/components/ReportActionItem/TripDetailsView.tsx
Comment (amyevans) [inline src/components/ReportActionItem/TripDetailsView.tsx:58]: If there's no `company` this is going to be an awkward string of spaces/dots, yeah?

File: src/libs/DateUtils.ts
Comment (amyevans) [reply src/libs/DateUtils.ts:772]: +1

File: src/libs/DateUtils.ts
Comment (amyevans) [inline src/libs/DateUtils.ts:787]: What does `mean` mean here? Maybe I'm sure reading it weirdly haha

File: src/libs/DateUtils.ts
Comment (amyevans) [inline src/libs/DateUtils.ts:790]:
```suggestion
 * 1. When the date refers to the current day: Departs on Sunday, Mar 17 at 8:00
 * 2. When the date refers not to the current day: Departs on Wednesday, Mar 17, 2023 at 8:00
```

File: src/types/onyx/Transaction.ts
Comment (amyevans) [inline src/types/onyx/Transaction.ts:303]: NAB but let's add a description here?

**Review by amyevans — CHANGES_REQUESTED** [review]
Few minor comments but overall LGTM!

File: src/libs/DateUtils.ts
Comment (rushatgabhane) [reply src/libs/DateUtils.ts]: nice catch! fixed it

File: src/libs/DateUtils.ts
Comment (rushatgabhane) [reply src/libs/DateUtils.ts]: haha i don't even remb. simplified the comment

File: src/types/onyx/Transaction.ts
Comment (rushatgabhane) [reply src/types/onyx/Transaction.ts]: Turns out it was dead code. Was from initial implementation. 
Removed it

File: src/components/ReportActionItem/TripDetailsView.tsx
Comment (rushatgabhane) [reply src/components/ReportActionItem/TripDetailsView.tsx]: nice catch! resolved for all

**Comment by rushatgabhane** [comment]
Screenshot 2024-05-30 at 12 44 30

**Comment by rushatgabhane** [comment]
Requesting a re-review : )

File: src/styles/index.ts
Comment (shawnborton) [inline src/styles/index.ts:416]: Are you sure we don't have an existing style for this? We use bold text all throughout the app. Let's try to avoid writing styles if there is a style that exists that we can reuse.

**Comment by shawnborton** [comment]
What's the difference between this PR and [this PR here](https://github.com/Expensify/App/pull/38808)? I left a bunch of design feedback in the linked PR, just want to make sure it's taken care of.

**Comment by rushatgabhane** [comment]
@shawnborton that's a different PR, it deals with preview of the room and this PR deals with the trip room itself. 

The other PR:

Screenshot_20240530-180939~2

File: src/styles/index.ts
Comment (rushatgabhane) [reply src/styles/index.ts:416]: Actually, I cant find something equivalent in the styles directory.

Comment (rushatgabhane) [reply src/styles/index.ts:416]: Maybe textStrong. Let me try that

```
textStrong: {
            fontFamily: FontUtils.fontFamily.platform.EXP_NEUE_BOLD,
            fontWeight: FontUtils.fontWeight.bold,
        }
```

**Comment by shawnborton** [comment]
Hmm but it seems like the styles I commented on in the other PR exist over here. Why is that?

**Comment by rushatgabhane** [comment]
we used the same base PR

**Comment by rushatgabhane** [comment]
@shawnborton you can do it on the other PR, I'll give it a look and push the suggestions in this PR too.

Would that work for us?

**Review by amyevans — DISMISSED** [review]
✈️ 🏨 🚗

**Comment by shawnborton** [comment]
Works for me, I already left plenty of feedback over there :)

**Comment by stitesExpensify** [comment]
So to clarify @rushatgabhane are we holding this on the other PR? Or can this be merged without styles breaking in the future

**Comment by rushatgabhane** [comment]
@stitesExpensify after i incooperate @shawnborton's feedback in this PR, we can merge this : )

File: src/styles/index.ts
Comment (rushatgabhane) [reply src/styles/index.ts]: resolved!

**Comment by rushatgabhane** [comment]
@stitesExpensify all done! Ready for a review again

**Review by stitesExpensify — DISMISSED** [review]
LGTM! @shubham1206agra can we get another review here and the checklist please?

**Comment by rushatgabhane** [comment]
color used for icon is `border`

**Review by shubham1206agra — APPROVED** [review]

**Review by amyevans — DISMISSED** [review]

**Comment by dannymcclain** [comment]
@rushatgabhane @shubham1206agra Yup—the container around the icon should use our `border` color. The icon itself should be colored `icon`, and then of course the hover background should be `rowHover`.

CleanShot 2024-05-31 at 11 18 37@2x

**Comment by rushatgabhane** [comment]
@dannymcclain i just merged latest `main`, the colors now match the design.

image

**Comment by stitesExpensify** [comment]
@shawnborton mind taking a look and making sure you like the UI here? :)

**Comment by shawnborton** [comment]
Yup all good on my end, we can do another pass at design review in the other PR that is outstanding.

**Review by stitesExpensify — APPROVED** [review]
LGTM!

File: src/components/ReportActionItem/TripDetailsView.tsx
Comment (luacmartins) [inline src/components/ReportActionItem/TripDetailsView.tsx:139]: Just noticed this, but I think this will not trigger a re-render of the component if the transaction gets updated.

**Pull request merged**
